tests/test_queue.py

```diff
@@ -581,3 +582,190 @@ def test_workflow(self, var1: str) -> str:
 
     # Reset logging
     logging.getLogger("dbos").propagate = original_propagate
+
+
+def test_queue_recovery(dbos: DBOS) -> None:
+    step_counter: int = 0
+    queued_steps = 5
+
+    wfid = str(uuid.uuid4())
+    queue = Queue("test_queue")
+    step_events = [threading.Event() for _ in range(queued_steps)]
+    event = threading.Event()
+
+    @DBOS.workflow()
+    def test_workflow() -> list[int]:
+        assert DBOS.workflow_id == wfid
+        handles = []
+        for i in range(queued_steps):
+            h = queue.enqueue(test_step, i)
+            handles.append(h)
+        return [h.get_result() for h in handles]
+
+    @DBOS.step()
+    def test_step(i: int) -> int:
+        nonlocal step_counter
+        step_counter += 1
+        step_events[i].set()
+        event.wait()
+        return i
+
+    # Start the workflow. Wait for all five steps to start. Verify that they started.
+    with SetWorkflowID(wfid):
+        original_handle = DBOS.start_workflow(test_workflow)
+    for e in step_events:
+        e.wait()
+    assert step_counter == 5
+
+    # Recover the workflow, then resume it.
+    recovery_handles = DBOS.recover_pending_workflows()
+    event.set()
```

Comment:
Isn't there a chance that you get the recovery handles and call event.set() before the recovered workflows executes?